### PR TITLE
Separate Configuration Settings per System Console section

### PR DESCRIPTION
**Pull request by cwarnermm**
Continuing to split out configuration settings into separate pages based how they're grouped via the System Console.

### CODE REVIEW

File: source/configure/configuration-settings.rst [hunk 1]
```diff
@@ -223,771 +223,81 @@ See the :doc:`Elasticsearch configuration settings </configure/elasticsearch-con
...
-
-*Available in legacy Enterprise Edition E20*
-
-The port used for streaming data between servers.
+See the :doc:`high availability configuration settings </configure/high-availability-configuration-settings>` documentation for details on the following configuration settings:
 
-+----------------------------------------------------------------------------------------------+
-| This feature's ``config.json`` setting is ``"StreamingPort": 8075`` with numerical input.    |
-+----------------------------------------------------------------------------------------------+
+- `Enable high availability mode <https://docs.mattermost.com/configure/push-notification-server-configuration-settings.html#enable-high-availability-mode>`__
```
Comment (justinegeffen) [inline source/configure/configuration-settings.rst:287]:
```suggestion
- `Enable High Availability mode <https://docs.mattermost.com/configure/push-notification-server-configuration-settings.html#enable-high-availability-mode>`__
```

File: source/configure/configuration-settings.rst [hunk 2]
```diff
@@ -5758,241 +5068,78 @@ Enable Global Retention Policy for Messages
...
 
-Go Routine Health Threshold
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+|enterprise| |cloud| |self-hosted|
 
-|all-plans| |self-hosted|
+*Available in legacy Enterprise Edition E20*
 
-This setting isn't available in the System Console and can only be set in ``config.json``.
+**True**: Files can be deleted as part of a scheduled data retention job. Doesn't apply to custom retention policies.
```
Comment (justinegeffen) [inline source/configure/configuration-settings.rst:5084]:
```suggestion
**True**: Files can be deleted as part of a scheduled data retention job. This doesn't apply to custom retention policies.
```

File: source/configure/high-availability-configuration-settings.rst [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+-----------------------------------------------------------+
+| You can enable high availability mode.                          | - System Config path: **Environment > High Availabilty**  |
+|                                                                 | - ``config.json`` setting: ``".ClusterSettings.Enable",`` |
+| - **true**: The Mattermost server will attempt inter-node       | - Environment variable: ``MM_CLUSTERSETTINGS_ENABLE``     |
+|   communication with the other servers in the cluster that      |                                                           |
+|   have the same cluster name. This sets the System Console to   |                                                           |
+|   read-only mode to keep the servers config.json files in sync. |                                                           |
```
[the remark this answers is not in the document]
Comment (justinegeffen) [reply source/configure/high-availability-configuration-settings.rst:52]: Note: The addition of one character may affect table alignment

File: source/configure/high-availability-configuration-settings.rst [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+Use gossip
+----------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+--------------------------------------------------------------------------------+
+| All cluster traffic uses the gossip protocol.                   | - System Config path: **Environment > High Availabilty**                       |
```
Comment (justinegeffen) [inline source/configure/high-availability-configuration-settings.rst:117]:
```suggestion
| All cluster traffic uses the gossip protocol.                   | - System Config path: **Environment > High Availability**                       |
```

File: source/configure/high-availability-configuration-settings.rst [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+Enable gossip compression
+-------------------------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+----------------------------------------------------------------------------------+
+| Once all servers in a cluster are upgraded to Mattermost v5.33  | - System Config path: **Environment > High Availabilty**                         |
```
Comment (justinegeffen) [inline source/configure/high-availability-configuration-settings.rst:158]:
```suggestion
| Once all servers in a cluster are upgraded to Mattermost v5.33  | - System Config path: **Environment > High Availability**                         |
```

File: source/configure/high-availability-configuration-settings.rst [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+
+Gossip port
+-----------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+---------------------------------------------------------------------+
+| The port used for the gossip protocol. Both UDP and TCP         | - System Config path: **Environment > High Availabilty**            |
```
Comment (justinegeffen) [inline source/configure/high-availability-configuration-settings.rst:177]:
```suggestion
| The port used for the gossip protocol. Both UDP and TCP         | - System Config path: **Environment > High Availability**            |
```

File: source/configure/high-availability-configuration-settings.rst [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+
+Streaming port
+--------------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+------------------------------------------------------------------------+
+| The port used for streaming data between servers.               | - System Config path: **Environment > High Availabilty**               |
```
Comment (justinegeffen) [inline source/configure/high-availability-configuration-settings.rst:191]:
```suggestion
| The port used for streaming data between servers.               | - System Config path: **Environment > High Availability**               |
```

File: source/configure/high-availability-configuration-settings.rst [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+Network interface
+-----------------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+------------------------------------------------------------------------+
+| An IP address used to identify the device that does automatic   | - System Config path: N/A                                              |
+| IP detection in high availability clusters.                     | - ``config.json`` setting: ``".ClusterSettings.NetworkInterface: "",`` |
```
Comment (justinegeffen) [inline source/configure/high-availability-configuration-settings.rst:219]:
```suggestion
| IP detection in High Availability clusters.                     | - ``config.json`` setting: ``".ClusterSettings.NetworkInterface: "",`` |
```

File: source/configure/high-availability-configuration-settings.rst [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+--------------------------------------------------------------------+
+| An IP address used to bind cluster traffic to a specific        | - System Config path: N/A                                          |
+| network device.                                                 | - ``config.json`` setting: ``".ClusterSettings.BindAddress: "",``  |
+|                                                                 | - Environment variable: ``MM_CLUSTERSETTINGS_BINDADDRESS``         |
+| This setting is used primarily for servers with multiple        |                                                                    |
+| network devices or different Bind Address and Advertise Address |                                                                    |
```
Comment (justinegeffen) [inline source/configure/high-availability-configuration-settings.rst:236]: Should Bind/Advertise/Address be capitalized?

File: source/configure/image-proxy-configuration-settings.rst [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+Enables image proxy
+--------------------
+
+|all-plans| |self-hosted|
+
+*Available in legacy Enterprise Edition E10/E20*
+
++---------------------------------------------------------------+---------------------------------------------------------------------+
+| An image proxy anonymizes Mattermost App connections and      | - System Config path: **Environment > Image Proxy**                 |
```
Comment (justinegeffen) [inline source/configure/image-proxy-configuration-settings.rst:42]:
```suggestion
| An image proxy anonymizes Mattermost app connections and      | - System Config path: **Environment > Image Proxy**                 |
```

File: source/configure/push-notification-server-configuration-settings.rst [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+*Available in legacy Enterprise Edition E10/E20*
+
++-----------------------------------------------------------------+--------------------------------------------------------------------------------+
+| Location of Mattermost Push Notification Service (MPNS),        | - System Config path: **Environment > Push Notification Server**               |
+| which re-sends push notifications from Mattermost to services   | - ``config.json setting``: ``".EmailSettings.PushNotificationServer",``        |
+| like Apple Push Notification Service (APNS) and Google Cloud    | - Environment variable: ``MM_EMAILSETTINGS_PUSHNOTIFICATIONSERVER``            |
+| Messaging (GCM).                                                |                                                                                |
+|                                                                 |                                                                                |
+| - Customers running an Enterprise or Professional Edition       |                                                                                |
```
Comment (justinegeffen) [inline source/configure/push-notification-server-configuration-settings.rst:62]:
```suggestion
| - Customers running a Professional or Enterprise Edition       |                                                                                |
```

File: source/configure/push-notification-server-configuration-settings.rst [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+|   Germany, enter ``https://hpns-de.mattermost.com/``.           |                                                                                |
+| - Team Edition customers should enter                           |                                                                                |
+|   ``https://push-test.mattermost.com``.                         |                                                                                |
++-----------------------------------------------------------------+--------------------------------------------------------------------------------+
+| **Note**:                                                                                                                                        |
+|                                                                                                                                                  |
+| - The TPNS is provided for testing push notifications prior to compiling your own service. Ensure you’re familiar with its `limitations          |
+|   <https://docs.mattermost.com/deploy/mobile-hpns.html#test-push-notifications-service-tpns>`__. Review the                                      |
```
Comment (justinegeffen) [inline source/configure/push-notification-server-configuration-settings.rst:73]: This link doesn't look like it's rendering.

File: source/configure/push-notification-server-configuration-settings.rst [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+| and @channel no longer send notifications to maximize           | - ``config.json setting``: ``".TeamSettings.MaxNotificationsPerChannel: 1000",``     |
+| performance.                                                    | - Environment variable: ``MM_EMAILSETTINGS_MAXNOTIFICATIONSPERCHANNEL``              |
+|                                                                 |                                                                                      |
+| Numerical input. Default is 1000.                               |                                                                                      |
++-----------------------------------------------------------------+--------------------------------------------------------------------------------------+
+| **Note**: We recommend increasing this value a little at a time, monitoring system health by tracking `performance monitoring metrics                  |
+| <https://docs.mattermost.com/scale/performance-monitoring.html>`__, and only increasing this value if large channels have restricted permissions       |
```
Comment (justinegeffen) [inline source/configure/push-notification-server-configuration-settings.rst:97]: Not sure if this link is rendering.

File: source/configure/web-server-configuration-settings.rst [hunk 14]
```diff
@@ -313,21 +313,203 @@ Purge all caches
...
+-------------------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++--------------------------------------------------------+-----------------------------------------------------------------------------------------+
+| Define the frequency, in milliseconds, of cluster      | - System Config path: N/A                                                               |
+| request time logging for performance monitoring.       | - ``config.json`` setting: ``".ServiceSettings.ClusterLogTimeoutMilliseconds: 2000",``  | 
+| for performanc monitoring                              | - Environment variable: ``MM_SERVICESETTINGS_CLUSTERLOGTIMEOUTMILLISECONDS``            |
```
Comment (justinegeffen) [inline source/configure/web-server-configuration-settings.rst:509]:
```suggestion
| for performance monitoring                              | - Environment variable: ``MM_SERVICESETTINGS_CLUSTERLOGTIMEOUTMILLISECONDS``            |
```

**Review by justinegeffen — COMMENTED** [review]
Some minor suggestions and a few comments about links that may not be rendering or it's just GitHub being weird. :) 
AWESOME WORK by the way!

File: source/configure/push-notification-server-configuration-settings.rst (hunk 12 above)
Comment (cwarnermm) [reply source/configure/push-notification-server-configuration-settings.rst:73]: Interesting. I've confirmed it's working locally and in the preview. I'll also double-check once these changes are merged into production.

File: source/configure/push-notification-server-configuration-settings.rst (hunk 13 above)
Comment (cwarnermm) [reply source/configure/push-notification-server-configuration-settings.rst:97]: Interesting. I've confirmed it's working locally and in the preview. I'll also double-check once these changes are merged into production.

File: source/configure/web-server-configuration-settings.rst [hunk 15]
```diff
@@ -313,21 +313,203 @@ Purge all caches
...
+Websocket URL
+-------------
+
+|all-plans| |self-hosted|
+
+*Available in legacy Enterprise Edition E10/E20*
+
++--------------------------------------------------------+---------------------------------------------------------------------+
+| You can configure the server to instruct clients       | - System Config path: N/A                                           |
+| on where they should try to connect websockets to.     | - ``config.json`` setting: ``".ServiceSettings.WebsocketURL: "",``  |                                                        | - Environment variable: ``MM_SERVICESETTINGS_WEBSOCKETURL``         |
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/configure/web-server-configuration-settings.rst:339]: Great catch! fixed!

File: source/guides/administration.rst [hunk 16]
```diff
@@ -23,6 +23,19 @@ Get started
     Web server configuration settings </configure/web-server-configuration-settings>
     Database configuration settings </configure/database-configuration-settings>
     Elasticsearch configuration settings </configure/elasticsearch-configuration-settings>
+    File storage configuration settings </configure/file-storage-configuration-settings>
+    Image proxy configuration settings </configure/image-proxy-configuration-settings>
+    SMTP configuration settings </configure/smtp-configuration-settings>
+    Push notification server configuration settings </configure/push-notification-server-configuration-settings>
+    High availability configuration settings </configure/high-availability-configuration-settings>
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/guides/administration.rst:30]: Leaving "high availability" as-is with no capitalization.

File: source/configure/configuration-settings.rst (hunk 1 above)
Comment (cwarnermm) [reply source/configure/configuration-settings.rst:287]: Leaving "high availability" as-is with no capitalization.

File: source/configure/configuration-settings.rst (hunk 2 above)
Comment (cwarnermm) [reply source/configure/configuration-settings.rst]: Fixed in most recent commit

File: source/configure/high-availability-configuration-settings.rst [hunk 17]
```diff
@@ -0,0 +1,298 @@
+.. _high-availability:
+
+High availability server configuration settings
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/configure/high-availability-configuration-settings.rst:3]: Leaving "high availability" as-is with no capitalization.

File: source/configure/high-availability-configuration-settings.rst [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+  :scale: 25
+  :target: https://mattermost.com/sign-up
+  :alt: Available for Mattermost Cloud deployments.
+
+.. |self-hosted| image:: ../images/self-hosted-badge.png
+  :scale: 25
+  :target: https://mattermost.com/deploy
+  :alt: Available for Mattermost Self-Hosted deployments.
+
+You can configure Mattermost as a `high availability environment <https://docs.mattermost.com/scale/high-availability-cluster.html>`__ by going to **System Console > Environment > High Availability**, or by editing the ``config.json`` file as described in the following table. Changes to configuration settings in this section require a server restart before taking effect.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/configure/high-availability-configuration-settings.rst:31]: Leaving "high availability" as-is with no capitalization.

File: source/configure/high-availability-configuration-settings.rst [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+
+.. tip::
+    In a Mattermost High Availability Cluster deployment, the System Console is set to read-only, and settings can only be changed by editing the ``config.json`` file directly. However, to test a High Availability environment, you can disable ``ClusterSettings.ReadOnlyConfig`` in the ``config.json`` file by setting it to ``false``. This allows changes applied using the System Console to be saved back to the configuration file.
+
+.. include:: common-config-settings-notation.rst
+    :start-after: :nosearch:
+
+Enable high availability mode
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/configure/high-availability-configuration-settings.rst:39]: Leaving "high availability" as-is with no capitalization.

File: source/configure/high-availability-configuration-settings.rst [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+
+Enable high availability mode 
+-----------------------------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+-----------------------------------------------------------+
+| You can enable high availability mode.                          | - System Config path: **Environment > High Availabilty**  |
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/configure/high-availability-configuration-settings.rst:47]: Really great catch!

File: source/configure/high-availability-configuration-settings.rst [hunk 21]
```diff
@@ -0,0 +1,298 @@
...
+
+Use IP address
+--------------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+------------------------------------------------------------------------+
+| You can configure your high availability environment to         | - System Config path: **Environment > High Availabilty**               |
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/configure/high-availability-configuration-settings.rst:100]: Leaving "high availability" as-is with no capitalization.

**Comment by cwarnermm** [comment]
@justinegeffen - fantastic feedback, as always! Would you be open to a quick second pass?

File: source/configure/configuration-settings.rst [hunk 22]
```diff
@@ -5754,245 +5064,82 @@ Enable Global Retention Policy for Messages
...
-TLS Cipher Overwrites
-^^^^^^^^^^^^^^^^^^^^^^
-
-|all-plans| |self-hosted|
-
-This setting isn't available in the System Console and can only be set in ``config.json``.
+*Available in legacy Enterprise Edition E20*
 
-Set TLS ciphers overwrites to meet requirements from legacy clients which don't support modern ciphers, or to limit the types of accepted ciphers.
+**True**: Messages can be deleted as part of a scheduled data retention job. Doesn't apply to custom retention policies.
```
Comment (justinegeffen) [inline source/configure/configuration-settings.rst:5069]:
```suggestion
**True**: Messages can be deleted as part of a scheduled data retention job. This doesn't apply to custom retention policies.
```

File: source/configure/deprecated-configuration-settings.rst [hunk 23]
```diff
@@ -107,8 +107,21 @@ Disable legacy MFA API endpoint
...
+------------------
+
+At Rest Encrypt Key
+~~~~~~~~~~~~~~~~~~~
+
+*Removed in August 23, 2018 release*
+
+This setting isn't available in the System Console and can only be set in ``config.json``. It's a legacy setting used to encrypt data stored at rest in the database, and no fields are encrypted using ``AtRestEncryptKey``. 
+
+A 32-character key for encrypting and decrypting sensitive fields in the database. When using High Availability, this value must be identical in each instance of Mattermost.
```
Comment (justinegeffen) [inline source/configure/deprecated-configuration-settings.rst:120]:
```suggestion
A 32-character key for encrypting and decrypting sensitive fields in the database. When using high availability, this value must be identical in each instance of Mattermost.
```

File: source/configure/high-availability-configuration-settings.rst [hunk 24]
```diff
@@ -0,0 +1,298 @@
...
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+------------------------------------------------------------+
+| You can enable high availability mode.                          | - System Config path: **Environment > High Availability**  |
+|                                                                 | - ``config.json`` setting: ``".ClusterSettings.Enable",``  |
+| - **true**: The Mattermost server will attempt inter-node       | - Environment variable: ``MM_CLUSTERSETTINGS_ENABLE``      |
+|   communication with the other servers in the cluster that      |                                                            |
+|   have the same cluster name. This sets the System Console to   |                                                            |
+|   read-only mode to keep the servers` ``config.json`` files     |                                                            |
```
Comment (justinegeffen) [inline source/configure/high-availability-configuration-settings.rst:52]:
```suggestion
|   read-only mode to keep the servers' ``config.json`` files     |                                                            |
```

File: source/configure/high-availability-configuration-settings.rst [hunk 25]
```diff
@@ -0,0 +1,298 @@
...
+
+Cluster name
+------------
+
+|enterprise| |self-hosted|
+
+*Available in legacy Enterprise Edition E20*
+
++-----------------------------------------------------------------+-----------------------------------------------------------------+
+| The cluster to join by name in a high availability environment. | - System Config path: **Environment > High Availabilty**        |
```
[the remark this answers is not in the document]
Comment (justinegeffen) [reply source/configure/high-availability-configuration-settings.rst:66]: Reverted caps; second instance is a typo fix.

File: source/configure/high-availability-configuration-settings.rst (hunk 21 above)
[the remark this answers is not in the document]
Comment (justinegeffen) [reply source/configure/high-availability-configuration-settings.rst:100]: Reverted caps; second instance is a typo fix.

File: source/configure/push-notification-server-configuration-settings.rst (hunk 13 above)
Comment (justinegeffen) [reply source/configure/push-notification-server-configuration-settings.rst:97]: I suspect it's just weird GitHub things, so thank you for checking!

**Review by justinegeffen — APPROVED** [review]
Awesome awesome work on this! Makes a huge difference. 

A few typo fixes but otherwise LGTM and approving.

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**